src/parser.rs

```diff
@@ -324,6 +371,48 @@ pub fn parse_gcov(gcov_path: &Path) -> Result<Vec<(String, CovResult)>, ParserEr
         .unwrap_or_else(|_| panic!("Failed to open gcov file {}", gcov_path.display()));
 
     let mut file = BufReader::new(&f);
+    if gcov_path.extension().unwrap() == "gz" {
+        let gz = GzDecoder::new(file);
+        let mut gcov: GcovJson = serde_json::from_reader(gz).unwrap();
```

Comment:
Maybe we should print a warning if the `format_version` is not the one we expect